numba/np/arraymath.py

```diff
@@ -669,6 +670,131 @@ def _trivial_isnan(x):
         return _trivial_isnan
 
 
+@register_jitable
+def _impl_iscomplex_isreal_arr(fn, x):
+    ax = np.asarray(x) # NumPy uses asanyarray here!
+    return fn(ax.imag, 0)
+
+
+@register_jitable
+def np_iscomplex_isreal(x, operator_fn, opt_ret):
+    if type_can_asarray(x):
+        return lambda x: _impl_iscomplex_isreal_arr(operator_fn, x)
+    elif isinstance(x, types.Optional):
+        return None
+    elif is_nonelike(x):
+        def impl(x):
+            return opt_ret
+    else:
+        def impl(x):
+            return False
```

Comment:
It seems odd to support `isreal("test") == False` unless numba already supports `np.array("test").imag == ""`.

Reply:
Numba doesn't support `np.asarray(any_string_here)`

Reply:
I've tried to implement `np.asarray("string")` [[PR](https://github.com/numba/numba/pull/5463)]. As long as the string is a literal, `np.asarray` works but when the string is promoted to `UnicodeType`, things become messy.

Therefore, I think the best path here is to limit the scope of this PR only to types `type_can_asarray` accepts and exclude strings. What do you think, @eric-wieser?

Reply:
l like the idea of only including types supported by `type_can_asarray`, especially since the numpy implementation essentially starts by calling `np.asarray`.